### DISCUSSION TITLE
Datalinks: Support for fetch() and other custom behaviors when clicking link

### DISCUSSION THREAD

**Discussion by maemigh** [grafana/grafana#38205] [Dashboards, Panels & Visualizations]
**What would you like to be added**:
Something like the check box for datalink to use the javascript fetch() call instead of opening in current/new window.

**Why is this needed**:
For calls to an API, I'm currently playing with having the link load and then using javascript to go back to the dashboard. Another thing I'm playing with is having the link open in a new window, and then using javascript in the new window to close itself.

It would be cleaner if fetch() was used to simply make the call to the API. I'm not too concerned about the status returned, but I guess it would be nice if this also showed some sort of loading indicator to the side of the datalink and then have different icons/badges appear and fade away to indicate success or failure.

**Comment by maemigh** [comment]
It would also be cool if the panel could be refreshed upon success from the datalink fetch(). In my case the API being hit will directly affect values represented in the panel the datalink is called from.

**Comment by torkelo** [comment]
sounds like you want a data link plugin type where you can do some custom behavior

**Comment by maemigh** [comment]
Yeah, that would be really cool

**Comment by maemigh** [comment]
I now have this working in a custom, hacked version of https://github.com/briangann/grafana-datatable-panel. However, this could be much better accomplished with https://github.com/grafana/grafana/issues/24482 as long as unsanitized (dangerous inner) HTML is allowed as well.

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
